### PR TITLE
Added initial demo scene with teleporters for scene-transitions.

### PR DESCRIPTION
**Pull request by Malcolmnixon**
Unregister signals from staging before fading to prevent processing duplicate scene-switch events.

Added meshes and supporting scenes/materials/shaders/scripts for mound, ramps, and teleport objects.
Added basic movement demo scene.
Added teleport from main menu level to basic movement demo scene.

### CODE REVIEW

File: assets/meshes/teleport/teleport.gd [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		return
+
+	# Teleport
+	if scene:
+		_scene_base.emit_signal("load_scene", scene.resource_path)
+	else:
+		_scene_base.emit_signal("exit_to_main_menu")
+
+func _set_title(value):
+	print("Setting title to ", value)
```
Comment (lunarcloud) [inline assets/meshes/teleport/teleport.gd:41]: Remove your print statements @Malcolmnixon

File: assets/meshes/teleport/teleport.gd [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	title = value
+	if is_inside_tree():
+		_update_title()
+
+func _update_title():
+	print("Updating title to ", title)
```
Comment (lunarcloud) [inline assets/meshes/teleport/teleport.gd:47]: Here too

**Review by lunarcloud — CHANGES_REQUESTED** [review]
You missed these

File: assets/meshes/teleport/teleport.gd (hunk 1 above)
Comment (Malcolmnixon) [reply assets/meshes/teleport/teleport.gd]: Good catch - left that debug print statement in there.

File: assets/meshes/teleport/teleport.gd (hunk 2 above)
Comment (Malcolmnixon) [reply assets/meshes/teleport/teleport.gd]: Fixed as well.

**Comment by Malcolmnixon** [comment]
I just pushed another update which adds a second demo scene for teleporting. This one also has the snap-turn provider in there which has two side-effects:
 1. The player has a body, and so can trigger the teleport-pads to leave
 2. The player has a body, and so can be affected by slopes and gravity

**Comment by BastiaanOlij** [comment]
Finally had a change to try this out. Very cool! I really like the teleporters!

**Review by BastiaanOlij — APPROVED** [review]

**Pull request merged**